### PR TITLE
Add generate new Text

### PR DESCRIPTION
**Pull request by MarcRaeder**

### CODE REVIEW

File: my-app/src/Components/GenerateNewText.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export async function GenerateNewText(textLength) {
```
Comment (DevBastiS) [inline my-app/src/Components/GenerateNewText.js]: You should stick to the 3 Layer Architecture here as well.
Meaning:

Create a Service and a Repository.
TextService and TextRepository.

The Repository is responsible for communicating with the API.
The Service handles Business Logic (nothing for this case I guess).

Create the Service and Repository and pass them as props to the component needing them.

File: my-app/src/Components/GenerateNewText.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  const fehlermeldung = "Es sind nur 1 - 100 Wörter erlaubt";
+  if (textLength < 1 || textLength > 100) {
+    return <div className="Fehlermeldung">{fehlermeldung}</div>;
+  }
+  let file = "https://random-word-api.herokuapp.com/word?number=" + textLength;
```
Comment (DevBastiS) [inline my-app/src/Components/GenerateNewText.js]: This should be the API you created in the previous assignment.

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: my-app/src/Service/TextService.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export class TextService {
+  constructor(repository) {
+    this.textRepository = repository;
+  }
+
+  async GenerateNewText(textLength) {
+    console.log(this.textRepository);
```
Comment (DevBastiS) [inline my-app/src/Service/TextService.js]:
```suggestion
```

File: my-app/src/Components/TypingArea.js [hunk 4]
```diff
@@ -1,16 +1,32 @@
...
 import { TextLengthInput } from "./TextLengtInput";
 import { TypingField } from "./TypingField";
+import { TextRepository } from "../Repository/TextRepository";
 
 export function TypingArea() {
+  const [text, setText] = useState("");
+  const [textLength, setTextLength] = useState(5);
+  const [textService, setTextService] = useState(new TextService(new TextRepository()));
+  useEffect(() => {
+    console.log(textService);
```
Comment (DevBastiS) [inline my-app/src/Components/TypingArea.js]:
```suggestion
```

File: my-app/src/index.css [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+  height: 100%;
+}
+
+#root {
+  height: 100%;
+  display: flex;
+}
+
+.TypingArea {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.typing-area {
```

Please adjust everywhere.

File: my-app/src/index.css [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  flex: 2;
+  background-color: blueviolet;
+}
+
+.StatsArea {
+  flex: 1;
+  background-color: green;
+}
+
+.TypingArea__Header {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.typing-area__header {
```

File: my-app/src/index.css [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  margin-top: 5%;
+}
+
+.TypingArea__input_error {
+  margin-left: 44%;
+  margin-top: 5%;
+  color: red;
+}
+
+.test {
```
Comment (DevBastiS) [inline my-app/src/index.css]: Remove if not needed.

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: my-app/src/index.css [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+.typing-area__header {
+  margin-top: 2%;
+  font-size: 40px;
+  text-align: center;
+  text-decoration: underline;
+}
+
+.typing-area__textLength {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.typing-area__text-length {
```

File: my-app/src/index.css [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+.typing-area__text--incorrect {
+  color: red;
+}
+
+.typing-area__text--current {
+  text-decoration: underline;
+  background-color: rgb(251, 247, 1);
+}
+
+.typing-area__buttonLine {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.typing-area__button-line {
```

File: my-app/src/index.css [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+  display: flex;
+  flex-direction: row;
+  margin-top: 2%;
+  justify-content: center;
+}
+
+.stats-area__buttonLine {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.stats-area__button-line {
```

File: my-app/src/index.css [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+  text-align: left;
+}
+
+.stats-area__stats {
+  margin-bottom: 5px;
+  margin-left: 20px;
+  text-align: left;
+}
+
+.stats-area__dataButton {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.stats-area__data-button {
```

File: my-app/src/index.css [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+  margin-top: 30%;
+}
+
+.stats-area__moreDetails {
```
Comment (DevBastiS) [inline my-app/src/index.css]:
```suggestion
.stats-area__more-details {
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

**Review by DevBastiS — APPROVED** [review]

**Review by stefanluth — APPROVED** [review]

**Pull request merged**